packages/app/cypress/e2e/runner/support/mochaEventsUtils.ts

```diff
@@ -169,10 +172,20 @@ export function runCypressInCypressMochaEventsTest<T> (snapshots: T, snapToCompa
   })
 
   const assertMatchingSnapshot = (win: Cypress.AUTWindow) => {
-    win.getEventManager().on('cypress:in:cypress:run:complete', (args: CypressInCypressMochaEvent[]) => {
-      const data = sanitizeMochaEvents(args)
-
-      bus.emit('assert:cypress:in:cypress', data)
+    return new Promise((resolve) => {
+      win.getEventManager().on('cypress:in:cypress:run:complete', (args: CypressInCypressMochaEvent[]) => {
+        resolve(sanitizeMochaEvents(args))
+      })
+    }).then((snapshot) => {
+      if (Cypress.env('SNAPSHOT_UPDATE') === 1) {
```

Comment:
should we read the snapshots here so we dont have to store it globally?

Reply:
Yeah, we could, I was trying fit in with the existing logic but it probably makes sense to change it.

Reply:
updated [289a01e](https://github.com/cypress-io/cypress/pull/28224/commits/289a01e759870652b60eecfdfcb0fe87042f0725)